common/lib/xmodule/xmodule/tabs.py

```diff
@@ -174,62 +169,104 @@ def from_json(tab_dict):
         Raises:
             InvalidTabsException if the given tab doesn't have the right keys.
         """
-        sub_class_types = {
-            'courseware': CoursewareTab,
-            'course_info': CourseInfoTab,
-            'wiki': WikiTab,
-            'discussion': DiscussionTab,
-            'external_discussion': ExternalDiscussionTab,
-            'external_link': ExternalLinkTab,
-            'textbooks': TextbookTabs,
-            'pdf_textbooks': PDFTextbookTabs,
-            'html_textbooks': HtmlTextbookTabs,
-            'progress': ProgressTab,
-            'static_tab': StaticTab,
-            'peer_grading': PeerGradingTab,
-            'staff_grading': StaffGradingTab,
-            'open_ended': OpenEndedGradingTab,
-            'notes': NotesTab,
-            'edxnotes': EdxNotesTab,
-            'syllabus': SyllabusTab,
-            'instructor': InstructorTab,  # not persisted
-            'ccx_coach': CcxCoachTab,  # not persisted
-        }
-
-        tab_type = tab_dict.get('type')
-        if tab_type not in sub_class_types:
+        available_tab_types = CourseTabManager.get_tab_types()
+        tab_type_name = tab_dict.get('type')
+        if tab_type_name not in available_tab_types:
             raise InvalidTabsException(
-                'Unknown tab type {0}. Known types: {1}'.format(tab_type, sub_class_types)
+                'Unknown tab type {0}. Known types: {1}'.format(tab_type_name, available_tab_types)
             )
+        tab_type = available_tab_types[tab_dict['type']]
+        tab_type.validate(tab_dict)
+        # TODO: don't import openedx capabilities from common
+        from openedx.core.djangoapps.plugins.api import CourseViewType
```

Comment:
I think the plugin interfaces shouldn't be in explicit django apps. Instead, they should be pure-python, and used by our django apps.